src/main/java/com/google/sps/data/DeliverySlotManager.java

```diff
@@ -34,159 +36,61 @@
-     */
-    public void setCoordinatesFromAddress(String startAddress) throws ApiException, InterruptedException, IOException, DataNotFoundException {
-      LatLng point = MapsRequest.getLocationFromAddress(startAddress);
-      if (point == null) {
-        // Geocoding API didn't find the coordinates corresponding to given address.
-        throw new DataNotFoundException("Unable to find address!");
-      }
-      this.latitude = point.lat;
-      this.longitude = point.lng;
-    }
-
-    /**
-     * Return a Date object from dateString.
-     */
-    private Date validateDate(String dateString, long startTimeInMiliseconds) throws BadRequestException {
-      SimpleDateFormat formatter = new SimpleDateFormat(dateFormat);
-      Date date = null;
-      try {
-        date = formatter.parse(dateString);
-      } catch (ParseException e) {
-        throw new BadRequestException("Please enter a valid date!");
-      }
-      // get the currentDate to check that date is not set for a past date
-      Date currentDay = new Date(System.currentTimeMillis());
-      if (date.compareTo(currentDay) < 0) {
-        throw new BadRequestException("Please enter a valid date!");
-      }
-      if (date.getTime() + startTimeInMiliseconds < currentDay.getTime()) {
-        throw new BadRequestException("Please enter a time in the future!");
-      }
-      return date;
-    }
-
-    /**
-     * Returns the integer value of numberString
-     */
-    private Integer parseInteger(String numberString) throws BadRequestException {
-      Integer number = null;
-      try {
-        number = Integer.parseInt(numberString);
-      } catch (Exception e) {
-        throw new BadRequestException(e.getMessage());
-      }
-      return number;
-    }
-
-    /**
-     * Returns the Double value of numberString
-     */
-    private Double parseDouble(String numberString) throws BadRequestException {
-      Double number = null;
-      try {
-        number = Double.parseDouble(numberString);
-      } catch (Exception e) {
-        throw new BadRequestException(e.getMessage());
-      }
-      return number;
-    }
-
-    /**
-     * Returns LocalTime object from timeString
-     */
-    private LocalTime parseTime(String timeString) throws BadRequestException {
-      LocalTime time = null;
-      DateTimeFormatter formatter = DateTimeFormatter.ofPattern(timeFormat);
-      try {
-        time = LocalTime.parse(timeString, formatter);
-      } catch (DateTimeParseException e) {
-        throw new BadRequestException(e.getMessage());
-      }
-      return time;
-    }
-
-    private long getNumberOfMiliseconds(LocalTime time) {
-      return (long)(time.getHour() * 3600 + time.getMinute() * 60) * 1000;
-    }
+  public DeliverySlot createDeliverySlot(Date deliveryDay, long startTimeInMiliseconds, long endTimeInMiliseconds, String userId, String address) throws ApiException, BadRequestException, DataNotFoundException, IOException, InterruptedException {
```

Comment:
nit: Consider making the responsibility of the caller to construct DeliverySlot however they see fit, rather than making that your responsibility, e.g.

public void createDeliverySlot(DeliverySlot slot);

You generally want to avoid being in the business of replicating someone else's constructors (although you can impose certain constraints here, e.g. provided deliverySlot must have a correct `startPoint` set and throw an exception if it doesn't).

If you did that you also didn't had to implement getDeliverySlotRequestEntity in this PR I think.

Reply:
I changed the method to `public void createDeliverySlot(DeliverySlot slot);` and now there is no need for it to be overloaded.